### PR TITLE
Return a unified time for all time function in one query

### LINKED ISSUE #1832: support transaction time
In order to implement DEFAULT CURRENT_TIMESTAMP,
we need to providing a transaction time, for each query, a uniform time can be obtained.

### PR DESCRIPTION
**Pull request by Astralidea**
Previously, ```select now(), utc_timestamp(), curdate()``` will returns three different time for every function.
The result is confusing. The pull request will return only a unified time for the three functions.
I suppose every statement is a transaction and only uses a transaction time as a result.

### CODE REVIEW

**Comment by imay** [comment]
Hi, @Astralidea 
You should fix this issue https://github.com/StarRocks/starrocks/issues/1566 in this CL.
And you also get time from this for `now()` evaluation and for GlobalVariable in Coordinator.

**Review by Seaven — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Comment by imay** [comment]
@Astralidea 
Code in Coordinator.java should be changed too.
Because BE will compute `now()` according to the value in TQueryGlobals.
So, you should set all the now_string in TQueryGlobals from this value.
```
this.queryGlobals.setNow_string(DATE_FORMAT.format(new Date()));
this.queryGlobals.setTimestamp_ms(new Date().getTime());
```

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -230,6 +230,7 @@ private void handleQuery() {
         StatementBase parsedStmt = null;
         try {
             ctx.setQueryId(UUIDUtil.genUUID());
+            ctx.resetTransactionStartTime();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java:233]: Why only update when handling query?

**Comment by Astralidea** [comment]
> @Astralidea Code in Coordinator.java should be changed too. Because BE will compute `now()` according to the value in TQueryGlobals. So, you should set all the now_string in TQueryGlobals from this value.
> 
> ```
> this.queryGlobals.setNow_string(DATE_FORMAT.format(new Date()));
> this.queryGlobals.setTimestamp_ms(new Date().getTime());
> ```

ok, new commit use startTime and this part of the logic is taken into account

**Review by Seaven — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java [hunk 2]
```diff
@@ -271,8 +275,10 @@ public Coordinator(Long jobId, TUniqueId queryId, DescriptorTable descTable,
         this.fragments = fragments;
         this.scanNodes = scanNodes;
         this.queryOptions = new TQueryOptions();
-        this.queryGlobals.setNow_string(DATE_FORMAT.format(new Date()));
-        this.queryGlobals.setTimestamp_ms(new Date().getTime());
+        long startTime = System.currentTimeMillis();
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java:278]: should pass now through the function argument like timezone.

**Comment by Astralidea** [comment]
run all

**Review by imay — APPROVED** [review]
LGTM
Please run daily test before merging

**Comment by Astralidea** [comment]
ok, I'll rebase and test

**Review by chaoyli — APPROVED** [review]

**Pull request merged**